staging/src/k8s.io/client-go/scale/client.go

```diff
@@ -213,3 +214,46 @@ func (c *namespacedScaleClient) Update(resource schema.GroupResource, scale *aut
 
 	return scaleObj.(*autoscaling.Scale), err
 }
+
+func (c *namespacedScaleClient) Patch(gvr schema.GroupVersionResource, name string, pt types.PatchType, data []byte) (*autoscaling.Scale, error) {
+	groupVer := gvr.GroupVersion()
+
+	// we need to set the API path based on GroupVersion (defaulting to the legacy path if none is set)
+	// TODO: we "cheat" here since the API path really only depends on group ATM, but this should
+	// *probably* take GroupVersionResource and not GroupVersionKind.
+	apiPath := c.client.apiPathResolverFunc(groupVer.WithKind(""))
```

Comment:
can we align logic and style with the update func?

Reply:
this already has the version (which is required for patch), so this inlined the relevant parts of pathAndVersionFor which Update calls